tests/unit/csv_parser_test.c

```diff
@@ -215,6 +215,27 @@ static void test_new_csv_reader_zd3151_ENT3023()
     SeqDestroy(list);
 }
 
+static void test_new_csv_reader_carriage_return()
+{
+    size_t len = 4;
+    const char *inputs[] =
+    {
+        "default,paths,path[awk],0,/bin/awk,source=agent\r\n",
+        "default,sys,ipv4_2[eth0],0,10.186,source=agent\r\n",
+        "default,mon,dev_temp3,0,0.00,source=agent\r\n",
+        "default,sys,ipv4_2[eth0],0,10.186,source=agent\r\n"
```

Comment:
@karlhto It seems you are missing the line which you mentioned in PR comment. Maybe change last line to be:

```
default,sys,ipv4_2[eth0],0,10.186,\r\n
```
(comma/empty field before carriage return).

A line with only quotes before carriage return could be good as well.